### PR TITLE
Close popovers on click outside or escape

### PR DESCRIPTION
**Pull request by jessegduke**
Limited changes to 2019 Quick Check functional areas, but could be leveraged in future patterns.

Foundational:  Added AlpineJS include in Main Layout template for 2019.  After full adoption of Alpine (if agreed via ADR), we should move into global level templates.

Functional changes:
**For _Share Results_ and _How was this calculated?_ helper popup**: Added functionality to close window upon: Escape or Enter keys and click outside.

**For "Help Me Prioritize**:    I did not close on Click outside because with the user already clicking various radio buttons, I did not want the user to "loose" their work upon an accidental click off modal.  However, I did add functionality to close window on "Escape" (a more deliberate action, in my opinon.

### CODE REVIEW

**Comment by davidstanke** [comment]
Preview this PR at https://draft-dora-dev--pr192-aeh8prsa.web.app

**Comment by davidstanke** [comment]
Thank you @jessegduke -- the functionality looks good! But we have a layout regression (see below). Can you address that, and also add a brief ADR to this PR? Thanks!

image

**Comment by davidstanke** [comment]
Preview this PR at https://draft-dora-dev--pr192-aeh8prsa.web.app

**Comment by jessegduke** [comment]
@davidstanke (Sorry I missed that).  Correction made for regression error in class/style for modal, and supplied brief ADR for JS Decorator proposal.  

Happy to adjust as needed.

File: hugo/content/quickcheck/2019/results.html [hunk 1]
```diff
@@ -18,88 +18,103 @@ <h3 ><span >Your performance:</span>
...
     <h3>You're performing better than <span id="percentile">&nbsp;&nbsp;&nbsp;</span>&percnt; of 2019 <a href="https://cloud.google.com/devops/state-of-devops/" target="_blank">State&nbsp;of&nbsp;DevOps&nbsp;Survey</a> respondents.&nbsp;</h3>
     <small>Take the Quick Check from other research years: <a href="/quickcheck/?year=2019">2019</a>&nbsp;&nbsp;<a href="/quickcheck/?year=2021">2021</a>&nbsp;&nbsp;<a href="/quickcheck/">2022</a></small>
     <br><br>
-    <button class="modal-open" aria-controls="share-modal">Share results</button>
+    <button x-on:click="isModalOpen = true" aria-controls="share-modal">Share results</button>
     <a href='{{% relref "/quickcheck/" %}}'><button class="secondary">Start over</button></a>
-    <div id="share-modal"  role="dialog" aria-modal="true" aria-labelledby="share-modal-label" aria-describedby="share-modal-content">
-        <div class="wrapper">
+    <div id="share-modal" x-show="isModalOpen" x-cloak x-transition role="dialog" aria-modal="true" aria-labelledby="share-modal-label" aria-describedby="share-modal-content">
+        <div x-cloak class="wrapper" x-on:click.away="isModalOpen = false">
```
Comment (davidstanke) [inline hugo/content/quickcheck/2019/results.html:35]: Here's where the magic happens. :)

**Review by davidstanke — APPROVED** [review]
Looks great -- and thank you for the ADR! Very well stated. 😎

**Pull request merged**